### PR TITLE
Content Pack Specific Files Validator

### PR DESCRIPTION
**Pull request by yardensade**
Don't let your mind be blown away by my PR naming.

## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19891
fixes: https://github.com/demisto/etc/issues/19886
fixes: https://github.com/demisto/etc/issues/19888

To whom it may concern:
@liorblob @yaakovi @Itay4 @jochman

### CODE REVIEW

**Comment by yardensade** [comment]
Creating tests atm, ready for PR

File: Tests/scripts/validate_files.py [hunk 1]
```diff
@@ -186,14 +187,19 @@ def get_modified_and_added_files(self, branch_name, is_circle, tag='origin/maste
...
-        return modified_files, added_files, old_format_files
+    @staticmethod
+    def get_packs(modified_files, added_files):
+        packs = set()
+        changed_files = modified_files.union(added_files)
+        for changed_file in changed_files:
+            pack = get_pack_name(changed_file)
+            if is_file_path_in_pack(changed_file) and pack:
+                packs.add(pack)
+        return packs
```
Comment (yaakovi) [inline Tests/scripts/validate_files.py]: empty line after loops

File: Tests/scripts/validate_files.py [hunk 2]
```diff
@@ -186,14 +187,19 @@ def get_modified_and_added_files(self, branch_name, is_circle, tag='origin/maste
...
+        packs = self.get_packs(modified_files, added_files)
 
-            # added_files = new_added_files
+        return modified_files, added_files, old_format_files, packs
 
-        return modified_files, added_files, old_format_files
+    @staticmethod
+    def get_packs(modified_files, added_files):
+        packs = set()
+        changed_files = modified_files.union(added_files)
```
Comment (yaakovi) [inline Tests/scripts/validate_files.py]: `modified_files` may contain tuples and not only strings

File: Tests/scripts/validate_files.py [hunk 3]
```diff
@@ -186,14 +187,19 @@ def get_modified_and_added_files(self, branch_name, is_circle, tag='origin/maste
...
+        for changed_file in changed_files:
+            pack = get_pack_name(changed_file)
+            if is_file_path_in_pack(changed_file) and pack:
```
Comment (yaakovi) [inline Tests/scripts/validate_files.py]:
```suggestion
            if pack and is_file_path_in_pack(changed_file):
```

File: Tests/scripts/hook_validations/pack_unique_files.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    def _is_pack_file_exists(self, file):
+        """Check if .secrets-ignore exists"""
+        if not os.path.isfile(self._get_pack_file_path(file)):
+            self._add_error('"{}" file does not exist, create one in the root of the pack'.format(file))
+            return False
+
+        return True
+
+    def _open_and_read_file_content(self, file_name):
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/pack_unique_files.py]: redundant, you can't read without open
```suggestion
    def _read_file_content(self, file_name):
```

File: Tests/scripts/hook_validations/pack_unique_files.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            errors = '@@@Issues with unique files in pack: {}\n  {}'.format(self.pack, '\n  '.join(self._errors))
+
+        return errors
+
+    # file utils
+    def _get_pack_file_path(self, file_name=''):
+        """Returns the full file path to pack's file"""
+        return os.path.join(self.pack_path, file_name)
+
+    def _is_pack_file_exists(self, file):
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/pack_unique_files.py]: don't shadow builtin function
```suggestion
    def _is_pack_file_exists(self, file_name):
```

**Comment by yardensade** [comment]
All goot on my end

**Pull request merged**